scanner/config/config.go

```diff
@@ -55,27 +63,86 @@ func (c *Config) validate() error {
 	if c.GRPCListenAddr == "" {
 		return errors.New("grpc_listen_addr is empty")
 	}
+	if err := c.Indexer.validate(); err != nil {
+		return fmt.Errorf("indexer: %w", err)
+	}
+	if err := c.Matcher.validate(); err != nil {
+		return fmt.Errorf("matcher: %w", err)
+	}
 	return nil
 }
 
 // IndexerConfig provides Scanner Indexer configuration.
 type IndexerConfig struct {
 	// Database provides indexer's database configuration.
-	DBConnString string `yaml:"db_conn_string"`
+	Database Database `yaml:"database"`
 	// Enable if false disables the Indexer service.
 	Enable bool `yaml:"enable"`
 	// GetLayerTimeout timeout duration of GET requests for layers
 	GetLayerTimeout Duration `yaml:"get_layer_timeout"`
 }
 
+func (c *IndexerConfig) validate() error {
+	if !c.Enable {
+		return nil
+	}
+	if err := c.Database.validate(); err != nil {
+		return fmt.Errorf("database: %w", err)
+	}
+	return nil
+}
+
 // MatcherConfig provides Scanner Matcher configuration.
 type MatcherConfig struct {
 	// Database provides matcher's database configuration.
-	DBConnString string `yaml:"db_conn_string"`
+	Database Database `yaml:"database"`
 	// Enable if false disables the Matcher service and vulnerability updater.
 	Enable bool `yaml:"enable"`
 }
 
+func (c *MatcherConfig) validate() error {
+	if !c.Enable {
+		return nil
+	}
+	if err := c.Database.validate(); err != nil {
+		return fmt.Errorf("database: %w", err)
+	}
+	return nil
+}
+
+// Database provides database configuration for scanner backends.
+type Database struct {
+	// DBConnString provides database configuration.
+	ConnString string `yaml:"conn_string"`
+	// PasswordFile specifies the database password by reading from a file.
```

Comment:
Can you specify either here for in `ConnString` that it is only valid for the password to be specified in either one of these fields but not both? Maybe in the comment for the struct